docs/types/conversion.rst

```diff
@@ -8,16 +8,19 @@ Conversions between Elementary Types
 Implicit Conversions
 --------------------
 
-If an operator is applied to different types, the compiler tries to implicitly
-convert one of the operands to the type of the other (the same is true for assignments).
-This means that operations are always performed in the type of one of the operands.
+An implicit type conversion is automatically applied by the compiler in some cases
+during assignments, when passing arguments to functions and when applying operators.
 In general, an implicit conversion between value-types is possible if it makes
 sense semantically and no information is lost.
 
 For example, ``uint8`` is convertible to
 ``uint16`` and ``int128`` to ``int256``, but ``int8`` is not convertible to ``uint256``,
 because ``uint256`` cannot hold values such as ``-1``.
 
+If an operator is applied to different types, the compiler tries to implicitly
+convert one of the operands to the type of the other (the same is true for assignments).
+This means that operations are always performed in the type of one of the operands.
```

Comment:
Does the implicit type conversion applied by the compiler deterministic e.g., always convert to "higher" type? Asking because users may want to take measures if they know an implicit type conversion may produce incorrect results.

Reply:
I don't understand your question. Is it clarified by what is said some lines up or not?

Reply:
@bshastry I think implicit conversion is always safe, that is, the new type can hold all values of the previous type.

Reply:
I think it is not clarified because the lines above the diff describe which types may be implicitly convertible (e.g., `uint8` to `uint16`). This paragraph is talking about operators. So I'm wondering how binary operators (`+`, `-` etc.) perform implicit conversion.

If this is explained elsewhere, please ignore my comment.

Reply:
@leonardoalt okay, sounds good. please ignore my comment in that case.

Reply:
Maybe it should link to where it's explained?

Reply:
> In general, an implicit conversion between value-types is possible if it makes
sense semantically and no information is lost.	sense semantically and no information is lost.